src/client/datascience/variablesView/notebookWatcher.ts

```diff
@@ -32,6 +32,15 @@ export class NotebookWatcher implements INotebookWatcher {
     public get activeNotebook(): INotebook | undefined {
         return this.notebookEditorProvider.activeEditor?.notebook;
     }
+    public get activeNotebookExecutionCount(): number | undefined {
```

Comment:
This was a change that we needed along with not retaining context anymore. When we become visible, check if there is a currently active notebook and update the execution count in the variable view. Basically the same operation as if you switch active documents.